cmd/build.go

```diff
@@ -154,7 +158,69 @@ func picardImage() (string, error) {
 	return fmt.Sprintf("%s@%s", picardRepo, sha), nil
 }
 
+func configVersion(configBytes []byte) (string, bool) {
+	var raw map[string]interface{}
+	if err := yaml.Unmarshal(configBytes, &raw); err != nil {
+		return "", false
+	}
+
+	var configWithVersion struct {
+		Version string `yaml:"version"`
+	}
+	if err := mapstructure.WeakDecode(raw, &configWithVersion); err != nil {
+		return "", false
+	}
+	return configWithVersion.Version, true
+}
+
+func validateConfigVersionHelper(configFilename string) error {
+	invalidConfigError := "\nYou attempted to run a local build with version '%s' of configuration.\nLocal builds do not support that version at this time.\nYou can use 'circleci config process' to pre-process your config into a version that local builds can run (see 'circleci help config process' for more information)"
+
+	configBytes, err := ioutil.ReadFile(configFilename)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read config file")
+	}
+
+	version, isVersion := configVersion(configBytes)
+	if !isVersion || version == "" {
+		return errors.New("Unable to identify config version")
+	}
+
+	if version != "2.0" && version != "2" {
+		return fmt.Errorf(invalidConfigError, version)
+	}
+
+	return nil
+}
+
+func containsIdx(list []string, v1 string, v2 string) int {
+	for i, v := range list {
+		if v == v1 || v == v2 {
+			return i
+		}
+	}
+	return -1
+}
+
+func validateConfigVersion(args []string) error {
+	indexConfigFlag := containsIdx(args, "--config", "--c")
+	configPath := opts.configFilename
+
+	if indexConfigFlag > -1 && indexConfigFlag+1 < len(args) {
+		configPath = args[indexConfigFlag+1]
+	}
```

Comment:
I don't think we need to re-implement flag parsing. `spf13/pflag` (the flag library that is used by cobra) can be used to parse the `args`.

My suggestion would be:
* move `flags.StringVarP(&opts.configFilename, "config", "c", defaultConfigPath, "config file")` into an `addConfigFlag(opts proxyBuildArgs, flagset *pflag.FlagSet)`
* call `addConfigFlag` from `newLocalExecuteCommand()` to add it to the flag set used for documentation
* [create a new `pflag.FlagSet`](https://godoc.org/github.com/spf13/pflag#NewFlagSet). You can set ` ParseErrorsWhitelist.UnknownFlags = true` to ignore the other flags.
* create a new instance of `proxyBuildArgs{}` to hold the value from flags.
* call `addConfigFlag` from here (`validateConfigVersion`) to add the flag to the FlagSet
* call https://godoc.org/github.com/spf13/pflag#FlagSet.Parse on the new FlagSet, passing in `args`
* that will populate `opts.configFilename`